src/contexts/AppContext.js

```diff
@@ -23,8 +24,33 @@ const baseState = {
 const AppProvider = (props) => {
 
     const [state, dispatch] = useReducer(appReducer, {}, () => {
-        const localData = localStorage.getItem('app');
-        return localData ? {...baseState, ...JSON.parse(localData)} : baseState;
+        const appString = localStorage.getItem('app');
+        let localData;
+
+        if (appString) {
+            localData = JSON.parse(appString);
+        } else {
+            localData = {};
+        }
+
+        /*
+         * Override connection properties with those specified in the query
+         * string (if any) and default the SIP and websocket URIs.
+         */
+        new URLSearchParams(window.location.search).forEach((value, key) => {
+            localData[key] = value;
+        });
+
+        if (!localData.sipUri && localData.user) {
+            localData.sipUri = localData.user + "@" + window.location.hostname + ":5061";
+        }
+
+        if (!localData.serverWssUri && localData.sipUri) {
```

Comment:
I'm not 100% convinced of this, or that this is how it should get enabled.... but for now it seems valid to allow it. Maybe we'll add in a `mode=something` flag.... and do stuff based off of that. I'll have a think